hw02-DIYarrayList/src/main/java/hw02/MyArrayList.java

```diff
@@ -0,0 +1,230 @@
+package hw02;
+
+import java.lang.IndexOutOfBoundsException;
+import java.lang.UnsupportedOperationException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class MyArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 30;
+
+    private Object[] _array;
```

Comment:
```suggestion
    private Object[] array;
```

В java не принято начинать поля с подчеркивания.
Хотя это не криминал и вполне допустимо, если все поля называются в таком стиле, но у вас `numberOfElements` без подчеркивания.
Приведите к единому стилю.